pkg/verifier/notation/truststore.go

```diff
@@ -103,3 +107,10 @@ func (s trustStore) filterValidCerts(certs []*x509.Certificate) ([]*x509.Certifi
 	}
 	return filteredCerts, nil
 }
+
+// Namespaced verifiers could access both cluster-scoped and namespaced certStores.
+// But cluster-wide verifiers could only access cluster-scoped certStores.
+// TODO: current implementation always returns true. Check the namespace once we support multi-tenancy.
+func isCompatibleNamespace(_ context.Context, _ string) bool {
```

Comment:
curious why does this implementation live in the truststore.go? Is it functionality specific to truststore?

Reply:
I moved this function into keymanagementprovider.go and validate within `GetKeysFromMap` and `GetCertificatesFromMap`. I feel if we separate get and validation operations, it's likely to forget the pre-validation for developers without context.